### ISSUE TITLE
not happy with the new bugtracker 

### ISSUE THREAD

**Issue by oberon2007** [manjaro/release-plan#82]
After the first experiences with bugs.manjaro.org I must say that it does not appear to be really ideal.
While it has some fance new features, we now lose the linking between internal issues and with other github projects and also with our own code of the kind 'soandso mentioned this is dadada'.
Also it is a disadvantage that issues cannot be closed by commits and you cannot see when a commit is related to an issue.
No realtime updates, either.
Maybe we should re-think this again?
The guys at linux deepin have a dedicated github-repo called 'developer-center' for (more or less) centralized issue tracking. Of course issues are still opened in the per-package repos there, too. But since the comments are interlinked I find it easy to follow everything. Just similar to our release-plan issues ...

**Comment by oberon2007** [comment]
Also the system of notifications looks very inefficient to me, with the tons of messages about every change of status, resulting in about three emails for one factual action. On top of that it is difficult to even find the crucial information in these kinds of messages:
```
THIS IS AN AUTOMATED MESSAGE, DO NOT REPLY.

The following task has been changed. The changes are listed below. For full information about what has changed, visit the URL and click the History tab.

FS#14 - mhwd fails to set vbox driver with mt-1.13
User who did this: artoo (artoo)
Status: New -> Unconfirmed

More information can be found at the following URL:
bugs.manjaro.org/js/callbacks/quickedit.php/index.php?do=details&task_id=14

You are receiving this message because you have requested it from the Flyspray bugtracking system. If you did not expect this message or don't want to receive mails in future, you can change your notification settings at the URL shown above. 
```

**Comment by oberon2007** [comment]
( I miss the emoticons there, too ... :disappointed: )

**Comment by Ste74** [comment]
In addition of this i not receive any notification of new bugs, only bugs where i partecipate.. example i have received only new status for bugs #4...
Il 23/ago/2016 13:18, Bernhard Landauer <notifications@github.com> ha scritto:( I miss the emoticons there, too ... 😞 )

**Comment by oberon2007** [comment]
Yes, I noticed that, too.
And you cannot reply by email (I think?)

**Comment by Ste74** [comment]
Yes , only open link to view what happen

**Comment by udeved** [comment]
@philmmanjaro 

I think we need one parent category in flyspray, so we can move issues around.
If I look at project management, I can assign a parent container, we don't have them defined, only admin can do it.

**Comment by oberon2007** [comment]
This will not solve any of the above mentioned issues.

**Comment by oberon2007** [comment]
I think it is not good at all to have the bugtracker separate from github.

**Comment by udeved** [comment]
BS, how do you think big distros keep track og bugs? Not on github.

**Comment by oberon2007** [comment]
Don't use insulting attributes for my comments, please.

**Comment by oberon2007** [comment]
One more downside: You cannot display the closed issues like on github. It is really hard to follow how issues were addressed as soon as they are (considered) fixed. You have to know they were there and explicitly search for them.

**Comment by udeved** [comment]
@philmmanjaro replicated effectively github per repo structure.

I think, except for home page, all other categories should have a parent project, and the rest can be done with subcategories. We get only task list.

**Comment by oberon2007** [comment]
I think it was great to have bugs connected with the code on github. We can have an 'issue' repo on github. That's not really a parent were you can move categories around, but it can be treated similarly. Like that everything will remain interlinked.

**Comment by udeved** [comment]
Look bernhard, bugtrackers are standard equipment for bigger software projects. 
manjaro is bigger, with lots of subprojects, we need a centralized issue tracking, where we can reassign, move etc
github is nice, has nice features, but its not suitable for a project of this size that grows.
I posted on internal forum a markdown based tracker iirc, but I don't know if its good.

**Comment by oberon2007** [comment]
yes. maybe we just have not found the right thing yet, then? ... Above all I miss the links to the code, because it's such a great way to learn how issues are addressed and also a good way to change quickly if the fix really solves what my problem is.
As an example:
Yesterday you decided that an issue was fixed for you and closed it. I had no way of seeing what your fix was supposed to be. So I could only go through all the possibly relevant repos and check where you had committed at all, check the timestamps and all - tedious.
Then when I decided it was not fixed for me, I went back and had a hard time finding the old ticket at all, because they are not being displayed as mentioned above.
When I found it again, there was no reference to what you had done in the code.

**Comment by udeved** [comment]
We need to either play around in plyfpray to organize it better, current structure works too, or we look for a different tracker, that perhaps works together with github same way discourse does.
Hence possibly a markdown based tracker.

**Comment by oberon2007** [comment]
agreed. a github compatible bugtracker would be great from my pov.

**Comment by udeved** [comment]
https://github.com/bueltge/Markdown-for-MantisBT
https://www.mantisbt.org/index.php

**Comment by Ste74** [comment]
More bug trucker 
http://toppersworld.com/top-25-popular-bug-and-issue-tracking-softwares/
but not all free and not all with github plugin

**Comment by udeved** [comment]
I would test mantis if you don't know it. Very flexible and tons of plugins.

**Comment by oberon2007** [comment]
Mantis looks promising. Notes, history, associated diffs... Looks quite good to me! :+1:

**Comment by Ste74** [comment]
if you want to maybe change it is best done quickly before people get used to flyspray

**Comment by udeved** [comment]
I know mantis, I once came across it in a project that used it.
Its good enough I think.

**Comment by Ste74** [comment]
Maybe i voted myself :anguished:   :stuck_out_tongue_closed_eyes:

**Comment by oberon2007** [comment]
Sofar we don't have so many tickets on flyspray, yet. Maybe they can quite easily be ported.
As long as the url remains the same I support this. I hope the theming can also be changed a bit :stuck_out_tongue_winking_eye: I only saw their demo page...

**Comment by udeved** [comment]
This is what really kind of drives me nuts, that someone cares about theme in bugtracker.
You got to see it from a functional side, no room in development for schnickschnack.

**Comment by Ste74** [comment]
Ah ah @udeved=Spock  @oberon2007=mmm Kirk o McCoy :stuck_out_tongue_winking_eye:

**Comment by udeved** [comment]
I always like the guys all the others hated, the Romulans, cool ships.

**Comment by oberon2007** [comment]
I know @udeved :wink: And I do support the mantis idea, regardless of how it looks, as long as it does what it's supposed to do better. Absolutely.
It was just a side-note, because I think it will not hurt on top of the functionality if our "customers" who report the bugs think that it looks cool and professional. Also in my world it inspires productivity to work in a beautiful place rather than in an ugly or oldfashioned one... - but that's completely off-topic of course :stuck_out_tongue_winking_eye:

**Comment by udeved** [comment]
Well, we would have to see, I know mantis, but I don't know how well the markdown plugin works.
I find flyspray somewhat not intuitive to handle.

**Comment by philmmanjaro** [comment]
What about **Phabricator** which Sonar is [using](https://dev.solus-project.com/).

**Comment by udeved** [comment]
Lokks good too, but what about github plugin? Or do you plan to use it along with self hosting source code repos?

**Comment by udeved** [comment]
http://dcycleproject.org/blog/80/setting-phabricator-track-git-repo

**Comment by oberon2007** [comment]
Looks very promising. On the solus bugtracker you can even register with your github account. Being logged in I didn't even need a password :scream: :laughing: 
Looks great, very responsive and their reference list is quite impressive ...

**Comment by philmmanjaro** [comment]
flyspray as also the option to use serveral login options like github. I simply had not yet enabled all features.

**Comment by udeved** [comment]
Funnily from wikipedia:

Upon committing changesets to a source code repository, post-commit hooks within Git or Subversion can be configured to automatically notify MantisBT that changes have been made to the repository. GitHub is also capable of remotely notifying a MantisBT installation of changes to a source code repository. Another technique used for integrating MantisBT with web-based source code hosting providers is the use of a job scheduler such as cron to manually check for changes to a repository every few minutes, reporting any changes back to MantisBT.[11]

> SourceIntegration allows for issues within MantisBT to be automatically resolved when a changeset message contains special strings such as "Fixes #12345", which are configurable using regular expressions. This also allows for relationships to be automatically created between changesets and corresponding issues recorded in MantisBT. Multiple issue IDs can be specified in a single changeset message, although this is commonly discouraged as it indicates the changeset doesn't have a single, clear and logical purpose.[13]

Sound like what we need, except we want to host git repos under manjaro roof.

**Comment by oberon2007** [comment]
souds great!

**Comment by udeved** [comment]
@philmmanjaro 

What worries me a bit with phabricator, it targets professionals, perhaps not quite the right audience.

**Comment by philmmanjaro** [comment]
@oberon2007: I like **flyspray** cos it is simple. Even if a user created a bug in the wrong project we can move it. It takes the solved bugs out of the picture and people without knowledge can create an issue. What we have to do is create meaningful projects. The feature that you can close an issue with writing *fix #<some number> is only working with **github issues**. It is a feature. If I solve an issue I post the code section of github. Checkout my current work on the [MHWDCONFIG issue](https://bugs.manjaro.org/index.php?do=details&task_id=9). I see people opening tickets now. Before we all had it in our forums.

Github is easy if you develop stuff. It is easy to fork and interact with others. My plan is to close **github issues** on all of our repos and use **flyspray** as our only bugtracker. Every change is hard but you get use to it. If you don't like emails for the tickets you get assigned to, then remove the notification. If you like to get informed of all issues happening to a project you're involved you can use the rss feeds to get all notificiations for that project. **flyspray** has so many options which make sense. I set it especially to not bother you at all as you should use the options fit best for yourself.

It took me a while to find a simply bugtracker for people without any knowledge been able to post a problem they have. We might even create some project for installation issues. Give it time and you will see it. I'm working a long time with bugtrackers and software professionally. Create an issue for one issue and not multiple ones. Then everything can be solved one by one.

If you don't find the solution, then ask for the code line. Next time the author learns and post it automatically. If we want to have github accounts enabled, we can do that. Even facebook accounts are possible. I always thought, less is more so simple register via email is also fine. For me **flyspray** suits my needs completely.

**Comment by oberon2007** [comment]
Thank you @philmmanjaro That makes a lot of sense :)

**Comment by philmmanjaro** [comment]
@oberon2007: simply uncheck the option **Notify for own changes** so all changes you do yourself don't get mailed to you :smile: Others yes, if for example Marcus or I change the status of a change you are watching you get an email. If you don't like to get bothered about this issue you can uncheck yourself. It is called ` Watching: stop watching`. Maybe you didn't noticed that option yet.

Manuel might create a section in the beginners guide on how to post an issue. Simply checkout the tool and get used to it. Easy like that and **Enjoy the simplicity !!!!** :smile:

**Comment by philmmanjaro** [comment]
@Ste74: and not having the option to Email back forces you to check out what is happening. Simply mailing back without looking sometimes makes no sense at all. It will be more or less chatting in the end. If I'm responding I respond from the github page and not by email ...

**Comment by udeved** [comment]
I don't mind flyspray, its somewhat unintuitive managing stuff, but does the job.

@philmmanjaro 

Should we perhpas think about git hooks so to standardize commit stuff for repos?

**Comment by Ste74** [comment]
> oberon2007: simply uncheck the option Notify for own changes so all changes you do yourself don't get mailed to you :smile: Others yes, if for example Marcus or I change the status of a change you are watching you get an email. If you don't like to get bothered about this issue you can uncheck yourself. It is called **Watching: stop watching**. Maybe you didn't noticed that option yet.

But this work only for bug report created by own ? I don't see this option ...
and [here](https://bugs.manjaro.org/index.php?do=toplevel&project=0) for active rss ?

**Comment by jonathonf** [comment]
Ooch, I haven't been paying attention.

Phabricator is *awesome*. I have set up two instances now and it does *everything*, and it's really responsive. If what I've read is correct, it's in use at Facebook and is maintained by two ex-Facebook employees. Self-hosted setup is quite involved but you're fine if you actually read their installation instructions.

If you're after a pure bug-tracker, then MantisBT is fine, and we can also look at [YouTrack](https://www.jetbrains.com/youtrack/). We should be able to get either YouTrack self-hosted or hosted versions for free as we're already known to them as an open-source project.

If you want to go heavy-weight, then JIRA. Atlassian are happy to give free licenses to OSS projects.

**Comment by philmmanjaro** [comment]
Well, there is also now https://zammad.org/:

<iframe src="https://player.vimeo.com/video/136971025" width="640" height="360" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>

**Comment by oberon2007** [comment]
Biggest problem from my POV is still that people still don't get notified about new reports - I think it's only you, @philmmanjaro and probably artoo? Sometimes I visit the page to check if there's something that I can do but not very often, I just forget about it to be honest. Also what about @Ste74 and @Kirek ? I think they don't receive notifications, either...?

**Comment by oberon2007** [comment]
The only possible setting I seem to have in my profile is "Notify for own changes". Maybe I'm just still missing the switch where I could turn on those notifications?

**Comment by oberon2007** [comment]
Wow. I think, alas; now I have found something. You have to go to a specific project and then you will have a tab "manage project" where you can set notifications ...

**Comment by Ste74** [comment]
> Wow. I think, alas; now I have found something. You have to go to a specific project and then you will have a tab "manage project" where you can set notifications ...

I think no have this privilege 😉 i can't see this tab

**Comment by oberon2007** [comment]
When you are at the Tasklist (should be the startpage I think), on the top right you have a drop down menu showing "All projects". Select a project there. After that the top panel looks like that for me:
![bugtracker](https://cloud.githubusercontent.com/assets/11800714/21827606/6040b330-d784-11e6-817a-064c3e69a5d2.png)

**Comment by oberon2007** [comment]
It took me only 5 months to find that... :joy:

**Comment by Ste74** [comment]
Aha ... you have more privilege than me ... I think you have developer profile... for me is the basic profile.. I can't see nothing of your menu only the first :overview,tasklist and add new task.
